### DISCUSSION TITLE
特定キーが反応しない件について

### DISCUSSION THREAD

**Discussion by chikashimaru** [Yowkees/keyball#335] [Q&A]
既出の問題で大変申し訳ございませんが、
自力解決できそうにないので、ご助言をいただければ幸いです。

![IMG_0517のコピー](https://github.com/Yowkees/keyball/assets/8360319/2e5144d1-7f9b-4bae-953d-de99575bb8e6)

左手側でいくつか問題が発生しています。

1. 「S, V, X」の3つのキーが反応しない
キーソケットのはんだ付をやり直してみたり、該当するっぽいダイオードのはんだ付けを盛ってみたりしましたが、改善していません。。。改善に向けた手順がよくわかっていないので、ご助言いただけますと嬉しいです。

2. PRO Micro が反応しない？
最初は反応していたので、無事に左右両方のProMicroにファームウェアを書き込めました。ただ、上記のキー問題でいじっている間に、いつのまにか「PCと左ProMicro」を繋いでも反応しないようになりました。「PCと右ProMicro」は反応し、TRRSケーブルで左右のkeyballをつなぐと、左側のキーも上記の3キー以外は反応しています。
この点に関しても、なにかアドバイスいただけますと幸いです。

**Comment by chikashimaru** [comment]
その後、半田づけをやり直したりして、なんとか「V」「X」は反応するようになりました。
が、「S」だけはどうしてもうまくいかない・・・

**Comment by chikashimaru** [comment]
何度も申し訳ありません
1. 「S, V, X」の3つのキーが反応しない
こちらに関してはダイオードを付け替えて反応するようになりました！

2. PRO Micro が反応しない？
こちらに関して、ご助言いただけますと非常に嬉しいです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

反応しないキーについてはいろいろと対処いただき動くようにしていただきありがとうございます。

左側ProMicroにUSBケーブルを指しても認識しないということですが、OLEDの表示はどうなっていますでしょうか？
両手共にKeyballロゴが表示されている場合は正しくUSBデバイスとして認識できておらず動作しません。
ProMicroのType-Cコネクタには、ちょっとクセがあると思っておりまして、素早く奥まで差さないと認識されないことがあります。

左右のProMicroの入れ替えを試していただこうと思いましたが、半田付けされているようですので難しいですかね。
改善しないようでしたら、左手側のProMicro裏の半田付けで、隣とショートしていそうな個所がないかご確認いただければ幸いです。

以上、お手数ですがご対応お願いいたします。

**Reply by chikashimaru** [reply]
ご返信ありがとうございます。

左側ProMicroにUSBケーブルを指しても、OLEDには何も表示されない状態です。右側ProMicroに指すと、右OLEDには数字が、左OLEDにはロゴが表示されます。

左側ProMicroは、最初問題なく認識されていたのですが、キーが認識しない問題を解決しようと、左側（no ball siide）でテスターを色んなところに当てていました。その時、左側ProMicroとパソコンがUSBで繋がっていたのが悪かったのか？そのあたりのタイミングで接続がおかしくなった気がします。

ちなみに、ProMicroは裏からはんだ付けしなくてよかったんでしょうか？

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ProMicroは故障することがあるのでメンテナンス性のために、12Pコンスルーを使用しています。
コンスルーは各ピンがバネになっているので、半田付けしなくても保持力があります。
コネクタの抜き差しでコンスルーが外れるのを嫌う方は半田付けしてしまう方もみえます。

さて症状を見ると、左側ProMicroの故障も疑われるので左右のProMicroを入れ替えて確認したいところですが、取り外しできそうでしょうか？
お手数ですがご確認お願いいたします。

**Reply by chikashimaru** [reply]
なるほど。理解いたしました！！
左右のProMicro入れ替え、やってみます！
ちなみに、右ProMicroとPCをつなぐことで
今のところ問題なく使用できていますが、
この状態で使い続けるのはよろしくないのでしょうか？？

追伸＞まだまだkeyball44になれませんが、
日に日に自分にとって最高のキーボードになりつつあるのを感じます！！
ありがとうございます。

**Comment by Yowkees** [comment]
ぜひ操作に慣れて、最高のキーボードとなれれば幸いです！
おそらく左手側のProMicroコネクタ周辺の不具合かと思われますので、右側につないで問題なければこのままご使用いただいて大丈夫と思います。
今後ともどうぞよろしくお願いいたします。
